samples/sp-mgt/src/webparts/helloWorld/HelloWorldWebPart.ts

```diff
@@ -0,0 +1,75 @@
+import { BaseClientSideWebPart } from '@microsoft/sp-webpart-base';
+
+import styles from './HelloWorldWebPart.module.scss';
+
+import { Providers } from '@microsoft/mgt-element/dist/es6/providers/Providers';
+import { SharePointProvider } from '@microsoft/mgt-sharepoint-provider/dist/es6/SharePointProvider';
+import { PersonCardInteraction } from '@microsoft/mgt-components/dist/es6/components/PersonCardInteraction';
+import { PersonViewType } from '@microsoft/mgt-components/dist/es6/components/mgt-person/mgt-person-types';
+import { customElementHelper } from '@microsoft/mgt-element/dist/es6/components/customElementHelper';
+
+export default class HelloWorldWebPart extends BaseClientSideWebPart<{}> {
+  private _hasImportedMgtScripts = false;
+  private _errorMessage = '';
+
+  protected onInit(): Promise<void> {
+    Providers.globalProvider = new SharePointProvider(this.context);
+    customElementHelper.withDisambiguation('contoso');
+    return super.onInit();
+  }
+
+  public render(): void {
+    if (!this._hasImportedMgtScripts) {
```

Comment:
If this is meant to be done just once per web part, couldn't we move this to onInit? Also, what if there are multiple web parts in the solution, would we need a singleton for this to ensure that components with the same disambiguation are loaded only once?

Reply:
I guess we could, it's really a question of should we have an explicit call to render as a side effect in onInit? Having been out of the SPFx world for so long I don't know if that's a good practice or not.

Regarding the multiple webparts in the solution scenario, I don't think that we need a separate singleton to manage this behavior as the `customElementHelper` instance already provides a singleton for the solution, multiple calls to `withDisambiguation` have no additional effect provided the same value is used across all calls. So, I guess having a singleton to manage that call and ensure that it is called only once might be a good idea, or we could refactor the withDisambiguation to be non-rentrant and only accept the first truthy value that is received.

I'd love your thoughts on these options.

Reply:
> Looks really cool! From what I can see, with the default configuration it's backwards compatible, because the default prefix is `mgt-` but offers the ability to define a solution-specific prefix. Are there any drawbacks that you're aware of that we should into consideration?

Exactly, the idea is that the disambiguation call is completely optional and if an implementer chooses not to use it the ergonomics of the library remain unchanged from the current implementation.

As to drawbacks:
Increased bundle size for SPFx solutions vs using mgt-spfx
Increased complexity due to the need to load mgt-components asynchronously
Current implementation uses that last value provided to `withDisambiguation` which could create issues in multi web part solutions if differing values were supplied to `withDisambiguation`
> Should we refactor `withDisambiguation` to only accept the first truthy value and discard any others?

Reply:
> I guess we could, it's really a question of should we have an explicit call to render as a side effect in onInit? Having been out of the SPFx world for so long I don't know if that's a good practice or not.

Ah wait, I missed it. We can't call render from `onInit`, because onInit happens earlier in the lifecycle and there are other things to be done in between, so we should let SPFx runtime call render first.

> Regarding the multiple webparts in the solution scenario, I don't think that we need a separate singleton to manage this behavior as the customElementHelper instance already provides a singleton for the solution, multiple calls to withDisambiguation have no additional effect

That's great! We should definitely avoid unnecessary complexity where possible. Since we've got a reference to `render` in there, I think it makes sense to leave it where it is.

One thing that we could consider, is that since everyone using MGT would need this piece of code, perhaps we can look for ways to make it more pluggable rather than having everyone copy-paste this setup code in their project.

Reply:
@waldekmastykarz I'm not sure that we can offer much value in terms of a plugable approach for loading those scripts in a no framework webpart given that `render` and `renderError` are protected so can only be invoked by the webpart itself.
The most generic helper would be:
```typescript
export function importMgtComponentsLibrary(
  hasImportedMgtScripts: boolean,
  onSuccessCallback: () => void,
  onErrorCallback?: (e?: Error) => void
) {
  if (!hasImportedMgtScripts) {
    import('@microsoft/mgt-components')
      .then(() => {
        hasImportedMgtScripts = true;
        onSuccessCallback();
      })
      .catch(e => {
        onErrorCallback?.(e);
      });
  }
}
```
And the calling webpart still needs to supply the tracking boolean and the callback behaviors.
We'd also need to ship this in a package of its own due to `TS1323: Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'`

On the React side the helper would look like this:
```typescript
function lazyLoadComponent<P>(
  component: React.LazyExoticComponent<any>,
  props: P
): React.FunctionComponentElement<React.SuspenseProps> {
  return React.createElement(
    React.Suspense,
    {
      fallback: React.createElement('div', null, 'Loading')
    },
    React.createElement(component, props)
  );
}
```
And IMO delivers quite a bit more value

Reply:
I think it would be helpful to offer such functions and let developers call them as oneliners. If our logic changes at some point, we can update it without folks having to update their code, other than updating the dependency version.

Reply:
I'll add a new package to provide these spfx utilities. 
My biggest concern is that for the no framework case developers still need to have some internal state in the webpart that we're modifying in an external function.

Reply:
In this case, I would focus on React-based apps for SPFx. Let's see if the demand is there for no framework.